climada/util/coordinates.py

```diff
@@ -1305,38 +1472,55 @@ def _nearest_neighbor_euclidean(
     coordinates : 2d array
         First column contains latitude, second column contains longitude. Each
         row is a geographic point
+    unit : str
+        Unit to use for non-exact matching. Possible values are "degree", "m", "km".
```

Comment:
I think this is the point where the restriction to meters of kilometers would come into play, but I fail to see how. Could you please help me understand?